pkg/informer/supportedResources.go

```diff
@@ -37,11 +37,27 @@ func GetAllowDenyData(cm *v1.ConfigMap) ([]Resource, []Resource) {
 		Can't use configured value, defaulting to deny all resources. %v`, denyerr)
 	}
 
-	return allow, deny
+	return allow, deny, allowerr, denyerr
 }
 
 func isResourceAllowed(group, kind string, allowedList []Resource, deniedList []Resource) bool {
 
+	allowMap := GetAllowDenyDataMap(allowedList)
+	denyMap := GetAllowDenyDataMap(deniedList)
+
+	//case where same resources are in deny and allow list. expected behavior is to exclude those resources
+	// (default to deny) but we need to add a warning case
+	for _, allow := range allowMap {
+		for _, deny := range denyMap {
+			if (allow["group"] == deny["group"] && allow["rec"] == deny["rec"]) ||
+				(allow["group"] == "*" && deny["group"] == "*" && allow["rec"] == deny["rec"]) ||
+				(allow["rec"] == "*" && deny["rec"] == "*" && allow["group"] == deny["group"]) {
+				glog.V(2).Infof("Deny Resource [group: '%s' kind: %s]. Resource present in both allow and deny rule.", group, kind)
+				return false
+			}
+		}
+	}
```

Comment:
How about using this logic instead....

1. Add a new function that checks if a resource matches a resource list (abstracting allow/deny).
```go
function isResourceMatchingList(resourceList Resource[], group, kind string) bool{
  for _, r := range resourceList {
    for _, g := range r.ApiGroups {
	  for _, k := range r.Resources {
              if (g == "*" || g == group) && (k == "*" || k == kind) { // Group and kind matches
	        return false
              }
	  }
      }
    }
}
```

2. Then rewrite the original logic using the new function.
```go
func isResourceAllowed(group, kind string, allowedList []Resource, deniedList []Resource) bool {
  // Ignore clusters and clusterstatus resources because these are handled by the aggregator.
  // Ignore oauthaccesstoken resources because those cause too much noise on OpenShift clusters.
  // Ignore projects as namespaces are overwritten to be projects on Openshift clusters - they tend to share
  // the same uid.
  list := []string{"events", "projects", "clusters", "clusterstatuses", "oauthaccesstokens"}
  // Deny all apiResources with kind in list
  for _, name := range list {
    if kind == name {
	    glog.V(2).Infof("Deny resource [group: '%s' kind: %s]. Search collector doesn't support it.", group, kind)
	    return false
    }
  }

  // Deny resources that match the deny list.
  if isResourceMatchingList(denyList, group, kind) {
     // Check if resource is also in the allow list.
     if isResourceMatchingList(allowList, group, kind) {
        glog.V(2).Infof("Deny Resource [group: '%s' kind: %s]. Resource present in both allow and deny rule.", group, kind)
     } else {
        glog.V(2).Infof("Deny resource [group: '%s' kind: %s]. Matched rule [group: '%s' kind: %s].", group, kind, g, k) 
     }
     return false
  }

  // If allowList not provided, interpret it as allow all resources.
  // otherwise allow only the resources declared in allow list.
  if len(allowedList) == 0 {
    glog.V(2).Infof("Allow resource [group: '%s' kind: %s]. AllowList is empty.", group, kind)
    return true
  } else {
    if isResourceMatchingList(allowList, group, kind) {
      glog.V(2).Infof("Allow resource [group: '%s' kind: %s]. Matched [group: '%s' kind: %s].", group, kind, g, k)
    return true
    }
  }
  glog.V(2).Infof("Deny resource [group: '%s' kind: %s]. It doesn't match any allow or deny rule.", group, kind)
  return false
}

Reply:
Agree with this change. just minor change above we need to return the g and k from the isResourceMatchingList func.